EIPS/eip-6353.md

```diff
@@ -0,0 +1,333 @@
+}
+
+interface IERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///  uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///  `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID) external view returns (bool);
+}
+
+```
+
+To create and deploy an `ERC20Charity` contract one MAY only inherit directly from this [EIP-20](./eip-20.md) extension that would directly signal support for `ERC20Charity` :
+
+```solidity
+pragma solidity ^0.8.4;
+import "./ERC20Charity.sol";
+
+contract CharityToken is ERC20Charity {
+    // ...
+    constructor() ERC20("TestToken", "TST") { }
+    // ...
+}
+```
+
+checking if a contract implement this specification can be done with :
+
+```solidity
+//import "./IERC165.sol";
+import "@openzeppelin/contracts/interfaces/IERC165.sol";
+
+bytes4 private constant _INTERFACE_ID_ERCcharity = type(IERC20charity).interfaceId; // 0x557512b6
+
+function checkInterface(address testContract) external returns (bool) {
+    (bool success) = IERC165(testContract).supportsInterface(_INTERFACE_ID_ERCcharity);
+    return success;
+}
+```
+
+```solidity
+    mapping(address => uint256) public whitelistedRate; 
+    mapping(address =>  mapping(address => uint256)) private _donation; 
+    mapping (address =>address) private _defaultAddress; 
+
+    address[] whitelistedAddresses; //Addresses whitelisted
+
+    event AddedToWhitelist (address toAdd);
+    event RemovedFromWhitelist (address toRemove);
+    event DonnationAddressChanged (address whitelistedAddr);
+    event DonnationAddressAndRateChanged (address whitelistedAddr,uint256 rate);
+    event ModifiedCharityRate(address whitelistedAddr,uint256 rate)
+
+```
+
+### Functions
+
+#### **addToWhitelist**
+
+Add address to whitelist and set the rate to the default rate.
+
+| Parameter | Description |
+| ---------|-------------|
+| toAdd | The address to the whitelist.
+
+#### **deleteFromWhitelist**
+
+Remove the address from the whitelist and set rate to the default rate.
+
+| Parameter | Description |
+| ---------|-------------|
+| toRemove | The address to remove from whitelist.
+
+#### **getAllWhitelistedAddresses**
+
+Get all registered charity addresses.
+
+#### **getRate**
+
+Display for a user the rate of the default charity address that will receive donation.
+
+#### **setSpecificRate**
+
+Set personalized rate for charity address in {whitelistedRate}.
+
+| Parameter | Description |
+| ---------|-------------|
+| whitelistedAddr | The address to set as default. |
+| rate  | The personalised rate for donation. |
+
+#### **_returnRate**
+
+Return the rate to donate.
+
+| Parameter | Description |
+| ---------|-------------|
+| from | The address to get the rate of donation.
```

Comment:
This seems like an implementation detail, and it should be left out of the specification.